src/unix/raw.ml

```diff
@@ -65,33 +65,45 @@ let unsafe_read t ~off ~len buf =
   Stats.add_read n;
   n
 
+let assert_read ~len n =
+  assert (
+    if Int.equal n len then true
+    else (
+      Printf.eprintf "Attempted to read %d bytes, but got %d bytes instead!\n%!"
+        len n;
```

Comment:
Can you also print the backtrace?

Reply:
I'd rather enable backtrace collection in all tests, if we can do that: https://github.com/mirage/irmin/pull/1254

This way we take care of potential future hard-to-reproduce exceptions too.

Reply:
Great 👍